<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1282</width>
    <height>652</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="euclidAlg">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>131</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Euclidean Algorithm</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="euclidAlgOutput">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>90</y>
      <width>131</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="euclidAlgLeft">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="euclidAlgRight">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>60</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="modExp">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Mod Exponent</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="mExpBase">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>390</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="mExpExp">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>420</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="mExpMod">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="BASE">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>390</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>BASE</string>
    </property>
   </widget>
   <widget class="QLabel" name="EXPONENT">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>420</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>EXPONENT</string>
    </property>
   </widget>
   <widget class="QLabel" name="mExpModLabel">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>450</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>MODULUS</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="mExpOutput">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>490</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="MultInverse">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>30</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>MultInverse</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="mInvBase">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>80</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="mInvMod">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>110</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="mInvOut">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>140</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="mInvBaseLabel">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>80</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>BASE</string>
    </property>
   </widget>
   <widget class="QLabel" name="mInvModLabel">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>110</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>MOD</string>
    </property>
   </widget>
   <widget class="QLabel" name="euclidAlgA">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>A</string>
    </property>
   </widget>
   <widget class="QLabel" name="euclidAlgB">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>40</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>B</string>
    </property>
   </widget>
   <widget class="QPushButton" name="order">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>230</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Order</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="orderBase">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>290</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="orderMod">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>320</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="orderModLabel">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>320</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>MOD</string>
    </property>
   </widget>
   <widget class="QLabel" name="orderBaseLabel">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>290</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>BASE</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="orderOut">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>350</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="order_2">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>420</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Discrete Log</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="discreteLogMod">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>510</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="discreteLogValue">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>480</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="discreteLogPRoot">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>450</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="discreteLogModLabel">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>510</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>MOD</string>
    </property>
   </widget>
   <widget class="QLabel" name="discreteLogValueLabel">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>480</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>VALUE</string>
    </property>
   </widget>
   <widget class="QLabel" name="discreteLogPRootLabel">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>450</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>P ROOT</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="discreteLogOut">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>540</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="crtNumEqns">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>80</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="crtEqns">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>110</y>
      <width>171</width>
      <height>141</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="crtNumEqnsLabel">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>80</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>NUM EQUATIONS</string>
    </property>
   </widget>
   <widget class="QLabel" name="crtEqnFormat">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>120</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>EQN FORMAT:</string>
    </property>
   </widget>
   <widget class="QLabel" name="crtFormat1">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>140</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>NUM MOD NEWLINE</string>
    </property>
   </widget>
   <widget class="QLabel" name="crtFormat2">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>160</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>NUM MOD NEWLINE</string>
    </property>
   </widget>
   <widget class="QLabel" name="disclaimer">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>590</y>
      <width>391</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Note:  These WILL crash on improperly formatted input or input with no solution</string>
    </property>
   </widget>
   <widget class="QPushButton" name="crt">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>50</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>CRT</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionCalculateEuclideanAlg">
   <property name="text">
    <string>CalculateEuclideanAlg</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
